src/databricks/labs/ucx/source_code/jobs.py

```diff
@@ -194,10 +194,10 @@ def _register_spark_python_task(self, graph: DependencyGraph):
         if not self._task.spark_python_task:
             return []
         self._parameters = self._task.spark_python_task.parameters
-        notebook_path = self._task.spark_python_task.python_file
-        logger.info(f'Discovering {self._task.task_key} entrypoint: {notebook_path}')
-        path = self._as_path(notebook_path)
-        return graph.register_notebook(path, False)
+        python_file = self._task.spark_python_task.python_file
+        logger.info(f'Discovering {self._task.task_key} entrypoint: {python_file}')
+        path = self._as_path(python_file)
+        return graph.register_file(path)
```

Comment:
I believe we should check whether the file is a file or a notebook and register it accordingly. Notebooks need to be loaded by the NotebookLoader.

Reply:
Some context here is that I don't think notebooks are fully supported by this task interface, but on Monday I will perform some tests. (And maybe add some additional automated tests to confirm/explain the expected behaviour.)

My expectation here is that this is effectively a `spark-submit` interface and that it expects just a python file, effectively as if being run by the python interpreter without any of the notebook-specific handling. It's possible that a notebook might work, but only if the notebook is both python and has no magic because such a notebook can be run simply as a python script.

Reply:
My expectation was correct: this is a `spark-submit` interface that runs the associated file as a python script. It is not notebook-aware, so our linting needs to treat it the same way.